clients/src/main/java/org/apache/kafka/clients/consumer/RetriableCommitFailedException.java

```diff
@@ -16,21 +16,16 @@
  */
 package org.apache.kafka.clients.consumer;
 
-import org.apache.kafka.common.errors.RetriableException;
+import org.apache.kafka.common.errors.ApiException;
 
-public class RetriableCommitFailedException extends RetriableException {
+public class RetriableCommitFailedException extends ApiException {
```

Comment:
This is rather a fix: RetriableCommitFailedException should not be retriable (internally) as it indicate the caller whoever call commit-offsets that they need to retry. The only edge case is the auto commit mechanism, which calls commitOffsetAsync internally, but it should still be considered as a caller of that function so itself should be doing the retry logic.

Note this exception is already exposed as a public exception but there's no behavioral changes introduced by this.

Reply:
Just to clarify, the `RetriableException` marker interface refers to being internally retried, not to being retriable vs fatal for the external (user) caller?

Reply:
Yes.